### ISSUE TITLE
[DatePicker] Unable to select day with arrow keys once tabbed different month

### ISSUE THREAD

**Issue by marcusjwhelan** [mui/mui-x#4820]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Open any of the x-date-pickers. Tab to either a future or past month from the presented month and attempt to use arrow keys to select a day. This will not work, you can however open the DatePicker and then use the arrow keys on the current shown month to move forward or back to different months and select a day.

However this is not very usefully if you need to go several months ahead or back to select a day.

This is present on MobileDatePicker, DatePicker, and DesktopDatePicker.

StaticDatePicker has almost no use of arrow keys.

### Expected behavior 🤔

Open DatePicker, use Tab to change the month, use arrow keys to select the day.

### Steps to reproduce 🕹

Steps:

1. Go to Datepicker page and Tab to a DatePicker and hit enter to open
2. Tab to right arrow and hit enter to go forward a month
3. Attempt to use arrow keys to select day.

### Context 🔦

This is an accessibility issue with trying to use the keyboard to navigate to use the DatePicker intuitively.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
@emotion/react: 11.9.0 => 11.9.0
    @emotion/styled: 11.8.1 => 11.8.1
    @mui/base:  5.0.0-alpha.78
    @mui/icons-material: 5.6.2 => 5.6.2
    @mui/lab: 5.0.0-alpha.79 => 5.0.0-alpha.79
    @mui/material: 5.6.3 => 5.6.3
    @mui/private-theming:  5.6.2
    @mui/styled-engine:  5.6.1
    @mui/system:  5.6.3
    @mui/types:  7.1.3
    @mui/utils:  5.6.1
    @mui/x-date-pickers: 5.0.0-alpha.2 => 5.0.0-alpha.2
    @types/react: 18.0.8 => 18.0.8
    react: 18.1.0 => 18.1.0
    react-dom: 18.1.0 => 18.1.0
    typescript: 4.6.4 => 4.6.4
```
Firefox newest version
</details>

**Comment by alexfauquette** [comment]
Thanks for reporting, this issue. It seems similar to #4597

@marcusjwhelan could you close it if it's the same, or points which aspects are not covered by the other issue?
